### PR TITLE
Check message on construction

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/1254

verifying the structure of the message to make sure it follows the specs.
Three helper functions 
`errorCheckDelegationData`
`errorCheckMessageBody`
`errorCheckMessage`

## How to test:

yarn tests runs

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/actors_api/src/actors/Attester.spec.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ describe('Attester', () => {
     claimer = Identity.buildFromURI('//Bob')
 
     rawCType = {
-      $id: 'kilt:ctype:0x1',
+      $id: Crypto.hashStr('kilt:ctype:0x1'),
```
Comment (tjwelde) [inline packages/actors_api/src/actors/Attester.spec.ts:30]: why does this need to be hashed?

**Review by tjwelde — COMMENTED** [review]
Just a question about the ctypehash.
Is there maybe a bit of a inconsistency of how our ctype hash should look like?
I thought we wanted it to always follow the `kilt:ctype:` format...

File: packages/actors_api/src/actors/Attester.spec.ts (hunk 1 above)
Comment (Dudleyneedham) [reply packages/actors_api/src/actors/Attester.spec.ts:30]: It doesn't :)

Comment (Dudleyneedham) [reply packages/actors_api/src/actors/Attester.spec.ts:30]: It doesn't I have updated it accordingly.

File: packages/messaging/src/Message.spec.ts [hunk 2]
```diff
@@ -32,7 +31,7 @@ describe('Messaging', () => {
     const message = new Message(
       {
         type: Message.BodyType.REQUEST_CLAIMS_FOR_CTYPES,
-        content: [{ cTypeHash: '0x12345678' }],
+        content: [{ cTypeHash: `KILT:ctype:${Crypto.hashStr('0x12345678')}` }],
```
Comment (tjwelde) [inline packages/messaging/src/Message.spec.ts:34]: lowercase for `KILT`

File: packages/messaging/src/Message.spec.ts [hunk 3]
```diff
@@ -113,23 +112,18 @@ describe('Messaging', () => {
   })
 
   it('verifies the message sender is the owner', () => {
-    const content = {
-      claim: {
-        cTypeHash: '0x12345678',
+    const content = RequestForAttestation.fromClaimAndIdentity(
+      {
+        cTypeHash: `KILT:ctype:${Crypto.hashStr('0x12345678')}`,
```
Comment (tjwelde) [inline packages/messaging/src/Message.spec.ts:117]: same as above

**Review by tjwelde — CHANGES_REQUESTED** [review]
Looking good, just a quick casing issue

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**